test/src/main/java/org/apache/accumulo/test/FateSummaryIT.java

```diff
@@ -174,93 +187,32 @@ public void testFatePrintAndSummaryCommandsWithInProgressTxns() throws Exception
-
-        // create 250 txns each taking >= 20ms to complete >= 5 seconds total
-        for (int i = 0; i < numTxns; i++) {
-          // Initiate compaction to create txn. This compaction will take >= 20ms to complete
-          // ((10 key values) * (2ms sleep time / key value))
-          client.tableOperations().compact(table, null, null, false, false);
-        }
-
-        // Keep printing until we see a transaction complete mid-print or until we run out of
-        // transactions (they all complete).
-        // Realistically, should only take 1 or 2 iterations to see a transaction complete
-        // mid-print.
-        do {
-          // Execute the command when transactions are currently running and may complete mid-print
-          p = execAdminFateCommand(command);
-          // Previously, this check could fail due to a ZK NoNodeException
-          assertEquals(0, p.getProcess().waitFor());
-          result = p.readStdOut();
-          // A transaction should have completed mid-print and been ignored
-        } while (!result.contains("Tried to get info on a since completed transaction - ignoring")
-            && !noTransactions(result, command));
-
-        if (noTransactions(result, command)) {
-          // Fail since we printed until all transactions have completed and didn't see a
-          // transaction complete mid-print.
-          // This is highly unlikely to have occurred.
-          fail();
-        }
-        // Otherwise, we saw 'Tried to get info on a since completed transaction - ignoring', so
-        // test passes
-        client.tableOperations().delete(table);
-      }
-    }
-  }
-
-  private boolean noTransactions(String result, String command) {
-    if (command.equals("--print")) {
-      return result.contains(" 0 transactions");
-    } else { // --summary
-      result = result.substring(result.indexOf("{"), result.lastIndexOf("}") + 1);
-      FateSummaryReport report = FateSummaryReport.fromJson(result);
-      return report != null && report.getReportTime() != 0 && report.getStatusCounts().isEmpty()
-          && report.getStepCounts().isEmpty() && report.getCmdCounts().isEmpty()
-          && report.getStatusFilterNames().isEmpty() && report.getFateDetails().isEmpty();
-    }
-  }
-
-  private ProcessInfo execAdminFateCommand(String command) throws Exception {
-    if (command.equals("--print")) {
-      return getCluster().exec(Admin.class, "fate", command);
-    } else { // --summary
-      return getCluster().exec(Admin.class, "fate", command, "-j");
+    ServerContext sctx = getCluster().getServerContext();
+
+    // This error was occurring in AdminUtil.getTransactionStatus(). One of the methods that is
+    // called which may throw the NNE is top(), so we will mock this method to always throw a NNE
+    // and ensure it is handled/ignored within getTransactionStatus()
+    ZooStore<String> zs = EasyMock.createMockBuilder(ZooStore.class)
```

Comment:
Is this using partial mock?  I see the code in AdminUtil calls ZooStore.list and its not being mocked, so suspect partial mocking but not sure.

Reply:
Yes, using partial mocking here